src/plugins/intel_gpu/src/plugin/ops/kv_cache.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "intel_gpu/op/kv_cache.hpp"
+#include "intel_gpu/plugin/program_builder.hpp"
+#include "intel_gpu/plugin/common_utils.hpp"
+#include "intel_gpu/primitives/kv_cache.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+using KVCache = ov::intel_gpu::op::KVCache;
+}  // namespace internal
+}  // namespace op
+}  // namespace ov
+
+namespace ov {
+namespace intel_gpu {
+
+namespace {
+
+void CreateKVCacheOp(ProgramBuilder& p, const std::shared_ptr<ov::op::internal::KVCache>& op) {
+    validate_inputs_count(op, {2, 3});
```

Comment:
Currently it supports only 2 inputs - shouldn't we restrict option with 3 inputs?